### ISSUE TITLE
Null pointer exception 

### ISSUE THREAD

**Issue by arefeen** [shenkers/isoscm#12]
Hi,
I have run the IsoSCM (version 2.0.11) tool using the following command.

java -Xmx2048m -jar IsoSCM-2.0.11.jar assemble -bam simulated_mapped_read.sorted.bam -base simu_tool -s reverse_forward

But instead of giving output it gives the following error message:
Exception in thread "main" java.lang.NullPointerException
        at processing.FindSpliceJunctions.tabulateSpliceJunctions(FindSpliceJunctions.java:147)
        at executable.IsoSCM.main(IsoSCM.java:221)

Basically, the bam file that I am using is a small file that contains reads of some isoforms from chromosome 14 (of human). Am I doing anything wrong?
I can give you the bam file if you want. In that case, please give me your email address so that I can send the file to you.

**Comment by shenkers** [comment]
Hi @arefeen, it appears from your command you are using simulated reads, my guess is that you do not include an 'XS' attribute for the simulated spliced read alignments (this requirement is specified in the readme). The 'XS' attribute is either '+' or '-', and is assigned by spliced read aligners such as tophat that infer them from the polarity of splice junction motifs. Since you know the strand of the transcripts you are simulating you can add these tags during simulation. Alternatively you could map your simulated reads using an aligner that adds the XS tag. Something that has come up before is that if you are using STAR to align your reads, you need to add command line parameter '--outSAMstrandField intronMotif' to output the 'XS' tags for spliced reads.

Since I'm 99% certain it's an issue with the XS tags, I'm going to close this for now. Feel free to re-open if your spliced reads have XS tags, or you need more help with this.

**Comment by AMChalkie** [comment]
I also have a similar problem but with real data.

svi-4001:IsoSCM_3UTR achalk$ more isoscm/E861A-FL15wt.assembly_parameters.xml 
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<AssembleCommand>
    <bam>/Volumes/SVI_STEMCELL_BIOINFO/kallisto/Josh/./bpipe/results/E861A-FL15wt/merged/SM_E861A-FL15wt_RP_1_LB_005_ID_E861A-FL15wt_L0005_star-mm10.merged.dedup.sorted.bam</bam>
    <base>E861A-FL15wt</base>
    <coverage>true</coverage>
    <dir>/Volumes/SVI_STEMCELL_BIOINFO/ADAR/IsoSCM_3UTR/isoscm</dir>
    <internal>false</internal>
    <jnct_alpha>0.05</jnct_alpha>
    <merge_radius>100</merge_radius>
    <min_fold>0.5</min_fold>
    <min_terminal>300</min_terminal>
    <nb_r>10</nb_r>
    <segment_p>0.95</segment_p>
    <segment_r>10</segment_r>
    <strandedness>unstranded</strandedness>
    <threshold>1</threshold>
    <w>20</w>
</AssembleCommand>

java -Xmx2048m -jar $ISOSCMJAR assemble -bam /Volumes/SVI_STEMCELL_BIOINFO/kallisto/Josh/./bpipe/results/E861A-FL15wt/merged/SM_E861A-FL15wt_RP_1_LB_005_ID_E861A-FL15wt_L0005_star-mm10.merged.dedup.sorted.bam -base E861A-FL15wt -s reverse_forward
2016-12-06T16:15:01,824 [main] INFO  executable.IsoSCM - IsoSCM 2.0.13-SNAPSHOT
2016-12-06T16:15:01,900 [main] INFO  executable.IsoSCM - Command: java -jar -Xmx2048m IsoSCM-2.0.13-SNAPSHOT.jar assemble -bam /Volumes/SVI_STEMCELL_BIOINFO/kallisto/Josh/./bpipe/results/E861A-FL15wt/merged/SM_E861A-FL15wt_RP_1_LB_005_ID_E861A-FL15wt_L0005_star-mm10.merged.dedup.sorted.bam -base E861A-FL15wt -s reverse_forward
2016-12-06T16:15:01,989 [main] INFO  executable.IsoSCM - Starting assemble ...
2016-12-06T16:15:02,243 [main] INFO  executable.IsoSCM - tabulating splice junctions...
Exception in thread "main" java.lang.NullPointerException
	at processing.FindSpliceJunctions.tabulateSpliceJunctions(FindSpliceJunctions.java:148)
	at executable.IsoSCM.main(IsoSCM.java:240)

The bam file is STAR aligned, and I am testing the XS issue myself with https://groups.google.com/forum/#!topic/rna-star/t-oT4wUFFow however regardless of if this works or not it would be helpful for the software to return a more informative error message as the niave assumption is that you can run it on any bam file. Log files are also empty.

Thanks
/Alistair
